ReactiveCocoa/Swift/Signal.swift

```diff
@@ -680,11 +680,31 @@ public func tryMap<T, U, E>(operation: T -> Result<U, E>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` until `interval`. Then if signal isn't completed yet,
+/// errors with `error` on `scheduler`.
+public func timeoutWithError<T, E>(error: E, afterInterval interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
+
+	return Signal { observer in
+		let date = scheduler.currentDate.dateByAddingTimeInterval(interval)
+		let schedulerDisposable = scheduler.scheduleAfter(date) {
+			sendError(observer, error)
+		}
+
+		let signalDisposable = signal.observe(Signal.Observer { event in
+			observer.put(event)
```

Comment:
I don't think it's necessary to wrap `observer`. This could be `signal.observe(observer)`.